src/toil_lib/programs.py

```diff
@@ -15,34 +17,45 @@ def mock_mode():
     return True if int(os.environ.get('TOIL_SCRIPTS_MOCK_MODE', '0')) else False
 
 
-def docker_call(tool,
+def docker_call(job,
+                tool,
                 parameters=None,
                 work_dir='.',
                 rm=True,
+                detached=False,
                 env=None,
                 outfile=None,
                 inputs=None,
                 outputs=None,
                 docker_parameters=None,
                 check_output=False,
-                mock=None):
+                mock=None,
+                defer=None):
     """
     Calls Docker, passing along parameters and tool.
 
+    :param toil.Job.job job: The Job instance for the calling function.
     :param str tool: Name of the Docker image to be used (e.g. quay.io/ucsc_cgl/samtools)
     :param list[str] parameters: Command line arguments to be passed to the tool
     :param str work_dir: Directory to mount into the container via `-v`. Destination convention is /data
-    :param bool rm: Set to True to pass `--rm` flag.
+    :param bool rm: Should the container be run with the --rm flag (Should it be removed upon
+           container exit)? rm and detached are mutually exclusive in Docker.
```

Comment:
When referring to symbols in docstrings, you should always use backquotes around them. So 

```
`rm` and `detached`
```